site/publish.sh

```diff
@@ -1,26 +1,132 @@
 #!/bin/bash -ex
 
-# Used later for rsyncing updates
+## Environment variables that could be configured at the job level:
+# - OPT_IN_SYNC_FS_R2: (optional) Set it to "optin" to also update azure.updates.jenkins.io Files Share and R2 buckets
+
+## Variables declarations
+# For syncing updates.jenkins.io
 UPDATES_SITE="updates.jenkins.io"
 RSYNC_USER="www-data"
 
+# For syncing R2 buckets with aws-cli configured through environment variables (from Jenkins credentials)
+# https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+export AWS_DEFAULT_REGION="auto"
+UPDATES_R2_BUCKET="westeurope-updates-jenkins-io"
+UPDATES_R2_ENDPOINT="https://8d1838a43923148c5cee18ccc356a594.r2.cloudflarestorage.com"
+
+# For syncing Azure File Share
+UPDATES_FILE_SHARE_URL_AND_PATH="https://updatesjenkinsio.file.core.windows.net/updates-jenkins-io/"
+
+# For triggering a mirror scan on mirrorbits
+MIRRORBITS_POD_NAME_PREFIX="mirrorbits-lite"
+MIRRORBITS_CONTAINER_NAME="mirrorbits-lite"
+MIRRORBITS_NAMESPACE="updates-jenkins-io"
```

Comment:
```suggestion
```

Moving this instruction as close as where it is needed. Having all variables on top of a script makes it really hard to read and maintain: either "configuration" is managed separately or moved only "when" it is needed.